### ISSUE TITLE
Maybe a TreeSitter grammar?

### ISSUE THREAD

**Issue by StoneCypher** [StoneCypher/fsl#14]
https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/

We might be able to get out of it with the TextMate grammar though

https://flight-manual.atom.io/hacking-atom/sections/converting-from-textmate/

**Comment by StoneCypher** [comment]
Closing as a duplicate of #594 (Add a tree-sitter grammar), which is the concrete version of this speculative item. Continuing there.

**Comment by StoneCypher** [comment]
Reopening. (For the record: this wasn't closed by the recent automated backlog-triage pass — it was already closed as not_planned on 2026-05-31. Restoring it by maintainer request.)

The TreeSitter-grammar path is worth reviving rather than abandoning: tree-sitter has only become *more* central since this was filed — GitHub's own syntax highlighting and code navigation run on tree-sitter grammars, and Neovim, Zed, Helix and others use them as the primary highlighting mechanism. A tree-sitter grammar for FSL is the foundation for #15 (FSL as a recognized source + rendered language on GitHub), so this is the upstream prerequisite to unblock, not a dead end.

**Comment by StoneCypher** [comment]
Closing as a duplicate of #594 ("Add a tree-sitter grammar"), where tree-sitter tracking is being consolidated.

The TextMate-conversion shortcut mentioned here (TextMate grammar → tree-sitter / Atom) is folded into the broader "one canonical source → many grammars" strategy on #481. No content lost.
